pypeit/edgetrace.py

```diff
@@ -0,0 +1,4109 @@
+                viewer instead of a line and scatter plot.
+            include_img (:obj:`bool`, optional):
+                Overlay the trace data on the trace image; mutually
+                exclusive with `include_sobel`.
+            include_sobel (:obj:`bool`, optional):
+                Overlay the trace data on the Sobel-filtered trace
+                image; mutually exclusive with `include_img`.
+            img_buffer (:obj:`int`, optional):
+                Buffer for plot window with respect to trace image
+                size.
+            flag (:obj:`str`, :obj:`list`, optional):
+                One or more flags to use when *selecting* masked data
+                to include in the plot; unmasked data are always
+                plotted. If None, no masked data is plotted. If
+                'any', trace locations masked for any reason are
+                plotted. Otherwise, only trace locations flagged by
+                the specified bits are plotted.
+            idlabel (:obj:`bool`, optional):
+                Label each trace by their ID numbers.
+        """
+        if include_img and include_sobel:
+            msgs.error('Cannot show both the trace image and the filtered version.')
+        if in_ginga:
+            # Currently can only show in ginga if the edges are
+            # synchronized into slits
+            if not self.is_synced:
+                msgs.error('To show in ginga, slit edges must be left-right synchronized.')
+            ginga.connect_to_ginga(raise_err=True, allow_new=True)
+            # Show Image
+            viewer, ch = ginga.show_image(self.img, 'Trace Image')
+            if self.is_empty:
+                msgs.info('No traces defined.')
+                return
+            # Plot traces
+            # TODO: add masking
+            right = self.traceid > 0
+            cen = self.spat_cen if self.spat_fit is None else self.spat_fit
+            ginga.show_slits(viewer, ch, cen[:,np.invert(right)], cen[:,right],
+                             self.traceid[right], pstep=thin)
+            return
+
+        # Show the traced image
+        if include_img:
+            img_zlim = utils.growth_lim(self.img, 0.95, fac=1.05)
+            plt.imshow(self.img.T, origin='lower', interpolation='nearest', aspect='auto',
+                       vmin=img_zlim[0], vmax=img_zlim[1])
+        elif include_sobel:
+            sob_zlim = utils.growth_lim(self.sobel_sig, 0.95, fac=1.05)
+            plt.imshow(self.sobel_sig.T, origin='lower', interpolation='nearest', aspect='auto',
+                       vmin=sob_zlim[0], vmax=sob_zlim[1])
+
+        if self.is_empty:
+            msgs.info('No traces defined.')
+            plt.xlim(-img_buffer, self.nspec+img_buffer)
+            plt.ylim(-img_buffer, self.nspat+img_buffer)
+            plt.xlabel('Spectral pixel index')
+            plt.ylabel('Spatial pixel index')
+            plt.
```

Comment:
Maybe link to doc string of fully_masked_traces() so we can see what the
options are for this parameter.

Reply:
Done.